src/OneExplorer/OneExplorer.ts

```diff
@@ -229,6 +229,37 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
         });
   }
 
+  delete(oneNode: OneNode): void {
+    if (oneNode.node.type === NodeType.directory) {
+      vscode.window
+          .showInformationMessage(
+              `Are you sure you want to delete '${
+                  path.parse(oneNode.node.path).base}' and its contents?`,
+              {detail: `You can restore this file from the Trash.`, modal: true}, 'Move to Trash')
+          .then(ans => {
+            if (ans === 'Move to Trash') {
+              Logger.debug('OneExplorer', `Delete '${path.parse(oneNode.node.path).base}'.`);
+              return vscode.workspace.fs.delete(
+                  oneNode.node.uri, {recursive: true, useTrash: true});
+            }
+          })
```

Comment:
Q) `Balloon.info` has same structure with this code. But the `balloon`'s `showInformationMessage` promise pop up the message at right bottom of vscode window, and this code show the pop up a window which looks like OS-native. May I ask what's the difference..?

Reply:
may because of `modal: true` ?

Reply:
Please look into https://github.com/Samsung/ONE-vscode/pull/907#issuecomment-1171199706. The current delete command behavior looks alike the vscode-native.

Reply:
> maybe because of `modal: true` ?

This made the change! Thank you!!